pandas/core/internals/blocks.py

```diff
@@ -1850,10 +1850,39 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
-    def delete(self, loc) -> Block:
-        values = np.delete(self.values, loc, 0)
-        mgr_locs = self._mgr_locs.delete(loc)
-        return type(self)(values, placement=mgr_locs, ndim=self.ndim)
+    def delete(self, loc) -> list[Block]:
```

Comment:
Feels a bit strange to have a Block method now return a list of itself - is it possible to maintain the same return type while still getting a performance boost?

Reply:
Lots of Block methods return `list[Block]`

Reply:
Ah OK - sorry for the ignorance